improver_tests/synthetic_data/test_generate_metadata.py

```diff
@@ -401,31 +416,47 @@ def test_leading_dimension(cube_type, spp__relative_to_threshold):
 def test_set_attributes():
     """ Tests cube generated with specified attributes and the rest of the values set
     as default values """
-    attributes = {"source": "IMPROVER"}
-    cube = generate_metadata(attributes=attributes)
+    attributes = {"test_attribute": "kittens"}
+    cube = generate_metadata(MANDATORY_ATTRIBUTE_DEFAULTS, attributes=attributes)
 
+    expected_attributes = MANDATORY_ATTRIBUTE_DEFAULTS.copy()
+    expected_attributes["test_attribute"] = "kittens"
     assert cube.attributes == attributes
```

Comment:
I'm suprised this works, shouldn't it be == expected_attributes not attributes?

Reply:
Yes. It worked because `attributes` is being updated in the plugin. I've changed this so it takes a copy of that dict before modifying it. I've updated the unit-test to test the right thing too.